perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -1899,6 +1899,37 @@ public Application getApplicationById(PerunSession sess, int appId) throws Regis
 
 	}
 
+	@Override
+	public List<Application> getApplicationsForVo(PerunSession userSession, Vo vo, List<String> state, String dateFrom, String dateTo) throws PerunException {
+		vosManager.checkVoExists(userSession, vo);
+
+		//Authorization
+		if (!AuthzResolver.authorizedInternal(userSession, "getApplicationsForVo_Vo_List<String>_policy", Collections.singletonList(vo))) {
+			throw new PrivilegeException(userSession, "getApplicationsForVo");
+		}
+		if (state == null || state.isEmpty()) {
+			// list all
+			try {
+				return jdbc.query(APP_SELECT + " where a.vo_id=? and a.created_at::date >= date ? and a.create_at::date <= date ? order by a.id desc", APP_MAPPER, vo.getId(), dateFrom, dateTo);
+			} catch (EmptyResultDataAccessException ex) {
+				return new ArrayList<>();
+			}
+		} else {
+			// filter by state
+			try {
+				MapSqlParameterSource sqlParameterSource = new MapSqlParameterSource();
+				sqlParameterSource.addValue("voId", vo.getId());
+				sqlParameterSource.addValue("states", state);
+				sqlParameterSource.addValue("from", dateFrom);
+				sqlParameterSource.addValue("to", dateTo);
+				return namedJdbc.query(APP_SELECT + " where a.vo_id=:voId and state in ( :states ) and a.created_at::date >= date :from and a.create_at::date <= date :to order by a.id desc", sqlParameterSource, APP_MAPPER);
```

Comment:
This SQL seems to be invalid. If you would use the LocalDate, you can fix it like this
```suggestion
				return namedJdbc.query(APP_SELECT + " where a.vo_id=:voId and state in ( :states ) and a.created_at::date >= :from and a.created_at::date <= :to order by a.id desc", sqlParameterSource, APP_MAPPER);
```